recipes/openldap/all/conanfile.py

```diff
@@ -22,88 +27,73 @@ class OpenldapConan(ConanFile):
         "shared": False,
         "fPIC": True,
         "with_cyrus_sasl": True
-
     }
-    _autotools = None
-    _configure_vars = None
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        basic_layout(self, src_folder="src")
+
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def requirements(self):
-        self.requires("openssl/1.1.1q")
+        self.requires("openssl/3.2.1")
```

Comment:
```suggestion
        self.requires("openssl/[>=1.1 <4]")
```

In order to prevent not only version conflicts, but also downloading the latest version available, please, consider version range. 

https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges